pkg/network/usm/monitor.go

```diff
@@ -202,6 +210,10 @@ func (m *Monitor) Stop() {
 		return
 	}
 
+	if m.telemetryStopChannel != nil {
+		close(m.telemetryStopChannel)
+		m.telemetryStopChannel = nil
```

Comment:
When you set it to nil, the other goroutine can try to read from it when it's nil.

Reply:
addressed